### DISCUSSION TITLE
Slider widget to show deformed shape

### DISCUSSION THREAD

**Discussion by adhikarig** [pyvista/pyvista#3504] [Q&A] [answered]
Hello PyVista family,

I'm here to get your help. I use PyVista to visualise analysis models and post process the results from a structural FEM package OpenseesPy.  I was successful so far to visualise models and results at a particular time step.  Now I would like to add a slider widget so that I can run through the whole time series, similar to the chart overlays example.  When I use the slider widget, it keeps all instances instead of showing only single instance.  Here is a min working example.  Thanks in advance

```
pl = pv.Plotter()

nodes = [
    [0., 0., 0.],
    [0., 0., 2.],
    [0., 0., 4.]
]

deform = [
    [0., 0., 0.],
    [0.2, 0., 0.],
    [0.4, 0., 0.]    
]

lines = [[2, 0, 1],
         [2, 1, 2]
]

mesh_lines = pv.PolyData(nodes, lines=lines)
pl.add_mesh(mesh_lines, point_size=20, render=False)
pl.show(auto_close=False, interactive=True, interactive_update=True)

def create_mesh(value):      
    node_deform = nodes + value * np.array(deform)
    mesh_lines = pv.PolyData(node_deform, lines=lines) 
    _ = pl.add_mesh(mesh_lines, point_size=20, render=False)
    pl.update()
pl.add_slider_widget(create_mesh, [0, 100], 0, title='Resolution', event_type='always')
pl.show()
```

**Comment by banesullivan** [comment, marked as answer]
A quick fix is to use the `name` argument of `add_mesh` in your callback:

```py
_ = pl.add_mesh(mesh_lines, point_size=20, render=False, name='timestep_mesh')
```

The name argument is unique and will replace the mesh currently in the scene.

Hope this helps!

**Comment by banesullivan** [comment]
The other way to go about this is to only add the mesh to the scene once and then update the mesh in place

**Reply by banesullivan** [reply]
Your callback would look like:

```py
def create_mesh(value):      
    node_deform = nodes + value * np.array(deform)
    mesh_lines.copy_from(pv.PolyData(node_deform, lines=lines))  # used to be `overwrite`
    pl.update()
```
